pkg/datapath/node_addressing.go

```diff
@@ -36,12 +36,16 @@ type NodeAddressingFamily interface {
 	// on the node
 	AllocationCIDR() *cidr.CIDR
 
-	// LocalAddresses lists all local addresses
-	LocalAddresses() ([]net.IP, error)
+	// MapLocalAddresses lists all local addresses
```

Comment:
Nit: this godoc comment probably needs to be adjusted to reflect the map behavior, e.g.

```suggestion
	// MapLocalAddresses applies a function to each local address.
```
or similar.